ASM/c/debug.c

```diff
@@ -31,71 +38,217 @@ warp_t dungeon_warps[] = {
+};
+
+room_t forest_rooms[] = {
+    {  0x169, 0, { 110, 309, 641 }, 32768, "Entrance"},
+    {  0x169, 2, { 118, 383, -660 }, 32768, "Main Chamber"},
+    {  0x169, 11, { -1447, 383, -1439 }, 49152, "Block Puzzle room"},
+    {  0x169, 12, { -1328, 1228, -3326 }, 16282, "Red Poe"},
+    {  0x169, 13, { 586, 827, -3322 }, 16282, "Blue Poe"},
+    {  0x169, 15, { 1979, 403, -3404 }, 16282, "Checkerboard Ceiling"},
+};
+
+room_t fire_rooms[] = {
+    {  0x165, 0, { 5, 0, 846 }, 32768, "Entrance"},
+    {  0x165, 4, { 2989, 2060, 0 }, 49152, "Shortcut room"},
+    {  0x165, 5, { 2660, 2800, 0 }, 16384, "Boulder maze (down)"},
+    {  0x165, 5, { 1540, 2940, -630 }, 16384, "Boulder maze (up)"},
+    {  0x165, 8, { 1580, 4400, -509 }, 0, "Scarecrow room"},
+    {  0x165, 10, { -640, 2940, 190 }, 49152, "Fire wall maze"},
+    {  0x165, 13, { -2289, 4400, 19 }, 16384, "Hammer room"},
+};
+
+room_t water_rooms[] = {
+    {  0x10, 0, { -184, 780, 759 }, 32768, "Entrance"},
+    {  0x10, 17, { 987, 780, 181 }, 16384, "First ZL water level switch"},
+    {  0x10, 12, { -180, 0, -1870 }, 32768, "Pool of tektites and boulders"},
+    {  0x10, 8, { -1140, 60, -1810 }, 32768, "Dragon room"},
+    {  0x10, 13, { -3124, 1060, -1747 }, 32666, "Dark link room"},
+    {  0x10, 21, { -3113, 380, -4222 }, 4759, "River"},
+};
+
+room_t shadow_rooms[] = {
+    {  0x37, 2, { -254, -63, 594 }, 32768, "Entrance"},
+    {  0x37, 8, { 4164, -983, 1366 }, 49152, "Before huge room"},
+    {  0x37, 16, { 4677, -1143, 2474 }, 32768, "Invisible scythes"},
+    {  0x37, 10, { 1227, -1343, 3853 }, 49152, "Stone umbrella"},
+    {  0x37, 11, { 2477, -1343, 1393 }, 32666, "Invisible spikes"},
+    {  0x37, 21, { 4387, -1363, -1486 }, 32768, "Before boat"},
+    {  0x37, 15, { -3662, -1363, -1585 }, 49152, "Invisible wall maze"},
+};
+
+room_t spirit_rooms[] = {
+    {  0x82, 1, { -768, -50, -1 }, 49152, "After crawlspace (child)"},
+    {  0x82, 0, { 868, -50, 2 }, 16384, "After silver block (adult)"},
+    {  0x82, 15, { 1126, 480, -1281 }, 16384, "Spinning cobra room"},
+    {  0x82, 5, { 692, 480, -838 }, 49152, "Main room"},
+    {  0x82, 18, { 1787, 843, 152 }, 16282, "Armos room"},
+    {  0x82, 25, { 268, 1733, -828 }, 49152, "Ceiling mirror"},
+};
+
+room_t botw_rooms[] = {
+    {  0x98, 0, { 0, -12, 117 }, 32768, "Main room"},
+    {  0x98, 2, { -1650, 0, -739 }, 49152, "Coffin room"},
+    {  0x98, 3, { 1140, 0, -1339 }, 0, "Beamos room"},
+};
+
+room_t ice_rooms[] = {
+    {  0x88, 0, { 16, 0, 2678 }, 32768, "Entrance"},
+    {  0x88, 3, { 285, 38, 63 }, 32768, "Spinning scythe"},
+    {  0x88, 9, { 662, 52, -1656 }, 32768, "Worst room"},
+    {  0x88, 5, { -720, 60, -960 }, 49152, "Pushblock room"},
+    {  0x88, 6, { -1384, 277, 686 }, 8192, "Before wolfos room"},
+};
+
+room_t hideout_rooms[] = {
+    {  0x486, 2, { -296, 0, -2951 }, 16384, "Red cell 1 torch"},
+    {  0x486, 4, { -105, 0, -335 }, 16384, "Green cell 4 torches"},
```

Comment:
Slightly bothers me that these aren't in order

Reply:
This is the "vanilla order" you usually visit these rooms when playing Gerudo Fortress, so it made more sense to follow that rather than the number of torches.